### PR TITLE
Add new contributions permission

### PR DESCRIPTION
**Pull request by duartegalvao**
Adds a new `contributions` permission for managing event contributions

### CODE REVIEW

**Comment by ThiefMaster** [comment]
Looks good, one thing I'd test is whether the "undo" from the small popup you get when unassigning the session from an already-scheduled contribution (removing/changing the session unschedules id) still works. Because I think that might send a request to a timetable-related endpoint...

**Comment by ThiefMaster** [comment]
`RHAssignProgramCodesContributions` needs to either check the new permission as well or alternatively hide the "Assign program codes" button on the contributions list when you do not have full management permissions.

**Comment by ThiefMaster** [comment]
"Assign new session" and "Add new track" fail (as expected -> should be hidden).
Assigning an existing one works fine as expected.

**Comment by ThiefMaster** [comment]
Should the "publish/draft contributions" toggle be accessible with this permissions (currently it is), or should that require full management privileges? I'm leaning towards the latter since that goes beyond just managing (some) contributions...

**Comment by duartegalvao** [comment]
> Should the "publish/draft contributions" toggle be accessible with this permissions (currently it is), or should that require full management privileges? I'm leaning towards the latter since that goes beyond just managing (some) contributions...

Makes sense, I agree. For the jacow use-case all they need is to be able to edit titles/authors of contributions, so for this it wouldnt matter

**Comment by duartegalvao** [comment]
> "Assign new session" and "Add new track" fail (as expected -> should be hidden). Assigning an existing one works fine as expected.

I think I will just disallow any session/track assignment - imo those should be in a separate permission ("timetable" and "programme" respectively)

File: indico/modules/events/contributions/templates/management/_contribution_list.html [hunk 1]
```diff
@@ -168,40 +168,55 @@
...
+                                                        {% if contrib.session %}
+                                                            {{ contrib.session.title }}
+                                                        {% else %}
+                                                            {%- trans %}No session{% endtrans -%}
+                                                        {% endif %}
+                                                    </span>
+                                                    {% if not event.is_locked %}<span class="icon-arrow-down"></span>{% endif %}
+                                                </button>
+                                            {% elif contrib.session %}
+                                                    {{ contrib.session.title }}
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/templates/management/_contribution_list.html:188]: too much indentation

**Review by ThiefMaster — APPROVED** [review]

File: indico/modules/events/contributions/controllers/display.py [hunk 2]
```diff
@@ -168,7 +169,7 @@ class RHContributionJSON(RHContributionDisplayBase):
 
     def _process(self):
         from indico.modules.events.contributions.schemas import FullContributionSchema
-        can_manage = self.contrib.can_manage(session.user)
+        can_manage = self.contrib.can_manage(session.user, permission='contributions')
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/controllers/display.py:172]: That change wasn't correct, it's a management check on the contribution so there we need to check for full management access. Anyway, I reverted this in 986469c4d2.

```
ValueError: permission 'contributions' is not valid for 'Contribution' objects
```

**Pull request merged**